win-pvdrivers: unstable-2015-07-01 -> unstable-2023-08-17, win-signed-gplpv-drivers: remove

anthonyroussel opened the pull request:
## Description of changes

See https://xenproject.org/developers/teams/windows-pv-drivers/

* Updated source of win-pvdrivers
  * Upstream has been moved from https://github.com/ts468/win-pvdrivers to https://xenbits.xenproject.org/pvdrivers/win

* Added xencons, xenhid & xenxkbd drivers in win-pvdrivers
* Removed win-signed-gplpv-drivers package as it is now unmaintained upstream

  * Without real alternative because the Xen PV Drivers in win-pvdrivers are not signed
  * Xenproject further plans to release signed drivers, but in a future commercial offering
  * See https://help.univention.com/t/installing-signed-gplpv-drivers/21828

* Added myself as maintainer


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

anthonyroussel committed:
win-pvdrivers: unstable-2015-07-01 -> unstable-2023-08-17

ajs124 commented on pkgs/applications/virtualization/driver/win-pvdrivers/default.nix:
```diff
@@ -1,31 +1,75 @@
+      hash = "sha256-R8G5vG6Q4g0/UkA2oxcc9/jaHZQYb+u64NShCNt7s7U=";
+    }
+    {
+      name = "xenxkbd";
```
these `name` fields seem unused

anthonyroussel replied:
Fixed, thanks for the review

anthonyroussel commented:
@ofborg eval

anthonyroussel committed:
win-signed-gplpv-drivers: remove

win-signed-gplpv-drivers is now unmaintained upstream
See https://help.univention.com/t/installing-signed-gplpv-drivers/21828

Use win-pvdrivers instead (however, drivers are unsigned).

ajs124 commented:
Huh, what is ofborg up to? "Connection reset by peer"? Why would _that_ happen.

anthonyroussel commented:
> Huh, what is ofborg up to? "Connection reset by peer"? Why would _that_ happen.

Looks like [web.archive.org throttles the HTTP calls and may arbitrary closes the connection](https://github.com/brave/brave-browser/issues/3437#issuecomment-584234756) when an HTTP client makes too many requests to the web.archive.org server.
